src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -1516,14 +1516,45 @@ private static String getFieldTypeByFieldName(String name) {
         String fields = getFieldsAsString(anyStudy);
         String[] fieldsArray = name.split(DOT_SEPARATOR);
         String outerField = fieldsArray[OUTER_FIELD_INDEX];
-        String innerField = fieldsArray[INNER_FIELD_INDEX];
         Gson gson = new Gson();
         HashMap fieldsMap = gson.fromJson(fields, HashMap.class);
-        HashMap propertiesMap = gson.fromJson(String.valueOf(fieldsMap.get(PROPERTIES)), HashMap.class);
-        HashMap outerFieldsMap = gson.fromJson(String.valueOf(propertiesMap.get(outerField)), HashMap.class);
-        HashMap outerFieldPropertiesMap = gson.fromJson(String.valueOf(outerFieldsMap.get(PROPERTIES)), HashMap.class);
-        HashMap innerFieldMap = gson.fromJson(String.valueOf(outerFieldPropertiesMap.get(innerField)), HashMap.class);
-        return (String) innerFieldMap.get(TYPE);
+        HashMap propertiesMap = getField(gson, fieldsMap, PROPERTIES);
+        HashMap finalField = getField(gson, propertiesMap, outerField);
+        for (String field : Arrays.copyOfRange(fieldsArray, INNER_FIELD_INDEX, fieldsArray.length)) {
+            finalField = getFinalFieldDynamically(gson, finalField, field);
+        }
+        return (String) finalField.get(TYPE);
+    }
+
+    private static HashMap getFinalFieldDynamically(Gson gson, HashMap finalField, String field) {
+        return finalField.containsKey(PROPERTIES)
+                ? getPropertiesAndThenField(gson, finalField, field)
+                : getField(gson, finalField, field);
+    }
+
+    private static HashMap getField(Gson gson, HashMap finalField, String field) {
+        return gson.fromJson(String.valueOf(finalField.get(field)), HashMap.class);
+    }
+
+    private static HashMap getPropertiesAndThenField(Gson gson, HashMap finalField, String field) {
+        finalField = gson.fromJson(String.valueOf(finalField.get(PROPERTIES)), HashMap.class);
+        finalField = gson.fromJson(String.valueOf(finalField.get(field)), HashMap.class);
+        return finalField;
+    }
+
+    private static String getFieldTypeByArrayLength(String[] fieldsArray, String outerField, Gson gson, HashMap propertiesMap) {
+        if (fieldIsNested(fieldsArray)) {
+            String innerField = fieldsArray[INNER_FIELD_INDEX];
+            HashMap outerFieldsMap = gson.fromJson(String.valueOf(propertiesMap.get(outerField)), HashMap.class);
+            HashMap outerFieldsPropertiesMap = gson.fromJson(String.valueOf(outerFieldsMap.get(PROPERTIES)), HashMap.class);
+            return (String) gson.fromJson(String.valueOf(outerFieldsPropertiesMap.get(innerField)), HashMap.class).get(TYPE);
```

Comment:
is there chance that result of `outerFieldsPropertiesMap.get(innerField)` could be null? if so, calling `.get(TYPE)` will throw NPE